### PR TITLE
feat: added support for image editing and upscaling for imagen3

### PR DESCRIPTION
**Pull request by pavelgj**
```ts
  const baseImg = fs.readFileSync('base.png', { encoding: 'base64' });
  const maskImg = fs.readFileSync('mask.png', { encoding: 'base64' });

  const response = await generate({
    model: imagen3,
    output: { format: 'media' },
    prompt: [
      { media: { url: `data:image/png;base64,${baseImg}` }},
      {
        media: { url: `data:image/png;base64,${maskImg}` },
        metadata: { type: 'mask' },
      },
      { text: 'replace the background with foo bar baz' },
    ],
    config: {
      editConfig: {
        editMode: 'outpainting',
      },
    },
  });

  return response.media();
```

### CODE REVIEW

**Comment by MichaelDoyle** [comment]
See comments on #988.

**Comment by pavelgj** [comment]
> See comments on #988.

resolved

File: docs/plugins/vertex-ai.md [hunk 1]
```diff
@@ -157,6 +157,49 @@ const embedding = await embed({
...
+const response = await generate({
+  model: imagen3,
+  output: { format: 'media' },
+  prompt: 'a banana riding a bicycle',
+});
+
+return response.media();
+```
+
+and even advanved editing of exising images:
```
Comment (apascal07) [inline docs/plugins/vertex-ai.md:174]: Typo

**Review by apascal07 — APPROVED** [review]

File: docs/plugins/vertex-ai.md (hunk 1 above)
Comment (pavelgj) [inline docs/plugins/vertex-ai.md:174]:
```suggestion
and even advanced editing of existing images:
```

**Pull request merged**